### PR TITLE
Rules refactor

### PR DESCRIPTION
**Pull request by vvbandeira**
Add finer control to set when update rules:
- Update all rules independent of previous values.
- Update passing rules if they became tighter.
- Update failing rules.

Closes [OR-1325](https://openroad.atlassian.net/browse/OR-1325)

### CODE REVIEW

File: flow/util/genRuleFile.py [hunk 1]
```diff
@@ -1,220 +1,295 @@
...
 
-chdir(sys.argv[1])
-variant = sys.argv[2]
-okFile = f'metadata-{variant}-ok.json'
-outFile = f'rules-{variant}.json'
-rules = list()
+chdir(args.dir)
+metrics_file = f"metadata-{args.variant}-ok.json"
+old_rules_file = f"rules-{args.variant}.json"
+new_rules_file = f"rules-{args.variant}.json"
```
Comment (maliberty) [inline flow/util/genRuleFile.py]: Why are these the same?  It seems like you are just overwriting and could use the one variable.  Was this for testing?

File: flow/util/utils.mk [hunk 2]
```diff
@@ -15,7 +15,7 @@ update_metadata: $(REPORTS_DIR)/metadata-$(FLOW_VARIANT).json
 	      $(DESIGN_DIR)/metadata-$(FLOW_VARIANT)-ok.json
 
 update_rules:
-	$(UTILS_DIR)/genRuleFile.py $(DESIGN_DIR) $(FLOW_VARIANT)
+	$(UTILS_DIR)/genRuleFile.py $(DESIGN_DIR) --variant $(FLOW_VARIANT) --update
```
Comment (maliberty) [inline flow/util/utils.mk:21]: Change to -failing -tighten as discussed.

**Review by maliberty — APPROVED** [review]

File: flow/util/utils.mk (hunk 2 above)
Comment (vvbandeira) [reply flow/util/utils.mk:21]: Fixed on [17395ab](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/453/commits/17395ab20bf565656c2f980b0cf5148e3346fa87)

File: flow/util/genRuleFile.py (hunk 1 above)
Comment (vvbandeira) [reply flow/util/genRuleFile.py]: Fixed on [7490572](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/453/commits/74905727591051f695d8319bc6e512781f141e2a)

**Pull request merged**